### PR TITLE
stargate: split proto and base files

### PR DESCRIPTION
**Pull request by fedekunze**

### CODE REVIEW

File: scripts/contract-test.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
+!/bin/bash
```
Comment (araskachoi) [inline scripts/contract-test.sh]: improper shebang, need the `#`
change back to --> `#!/bin/bash`

File: scripts/integration-test-all.sh [hunk 2]
```diff
@@ -108,7 +108,7 @@ init_func() {
 
 start_func() {
     echo "starting ethermint node $i in background ..."
-    "$PWD"/build/ethermintd start --pruning=nothing --rpc.unsafe --log_level "main:info,state:info,mempool:info" \
+    "$PWD"/build/ethermintd start --pruning=nothing --rpc.unsafe \
```
Comment (araskachoi) [inline scripts/integration-test-all.sh:111]: are all the log_levels defaulted to `info` or is this logging unnecessary for the integration tests?

Comment (fedekunze) [reply scripts/integration-test-all.sh:111]: yeah need to fix this, good catch

File: scripts/proto-tools-installer.sh [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    Linux)
+        PROTOC_ZIP="protoc-${PROTOC_VERSION}-linux-x86_64.zip"
+        PROTOC_GRPC_GATEWAY_BIN="protoc-gen-grpc-gateway-v${PROTOC_GRPC_GATEWAY_VERSION}-linux-x86_64"
+    ;;
+    Darwin)
+        PROTOC_ZIP="protoc-${PROTOC_VERSION}-osx-x86_64.zip"
+        PROTOC_GRPC_GATEWAY_BIN="protoc-gen-grpc-gateway-v${PROTOC_GRPC_GATEWAY_VERSION}-darwin-x86_64"
+    ;;
+    *)
+        f_abort 1 "Unknown kernel name. Exiting."
```
Comment (araskachoi) [inline scripts/proto-tools-installer.sh:31]: ok to not have windows support? jw

File: scripts/run-solidity-tests.sh [hunk 4]
```diff
@@ -22,9 +20,7 @@ else
 fi
 
 chmod +x ./init-test-node.sh
-./init-test-node.sh > ethermintd.log &
-sleep 5
-ethermintcli rest-server --laddr "tcp://localhost:8545" --unlock-key localkey,user1,user2 --chain-id $CHAINID --trace --wsport 8546 > ethermintcli.log &
```
Comment (araskachoi) [inline scripts/run-solidity-tests.sh:27]: does the rest-server get automatically get deployed now with the new binary update? Or do we need to put back this command with the proper binary name?

also looks like `./init-test-node.sh` has references to `ethermintcli` which needs to be removed

File: scripts/run-solidity-tests.sh [hunk 5]
```diff
@@ -35,18 +31,14 @@ if (( $? != 0 )); then
...
 killall ethermintd
 
 echo "Script exited with code $ok"
 exit $ok
 
 # initializable-buidler fails on CI, re-add later
 
-./../../init-test-node.sh > ethermintd.log &
-sleep 5
-ethermintcli rest-server --laddr "tcp://localhost:8545" --unlock-key localkey,user1,user2 --chain-id $CHAINID --trace --wsport 8546 > ethermintcli.log &
```
Comment (araskachoi) [inline scripts/run-solidity-tests.sh:48]: same as above

File: scripts/start.sh [hunk 6]
```diff
@@ -1,5 +1,175 @@
...
+# "live" mode tests assume data dynamic
+
+SCRIPT=$(basename ${BASH_SOURCE[0]})
+TEST=""
+QTD=1
+SLEEP_TIMEOUT=5
+TEST_QTD=1
+
+#PORT AND RPC_PORT 3 initial digits, to be concat with a suffix later when node is initialized
+RPC_PORT="854"
```
Comment (araskachoi) [inline scripts/start.sh]: `8545` ?

File: scripts/start.sh [hunk 7]
```diff
@@ -1,5 +1,175 @@
...
+set -euxo pipefail
+
+DATA_DIR=$(mktemp -d -t ethermint-datadir.XXXXX)
+
+if [[ ! "$DATA_DIR" ]]; then
+    echo "Could not create $DATA_DIR"
+    exit 1
+fi
+
+DATA_CLI_DIR=$(mktemp -d -t ethermint-cli-datadir.XXXXX)
```
Comment (araskachoi) [inline scripts/start.sh:57]: should we rename this to something else?

File: third_party/proto/cosmos/auth/v1beta1/auth.proto [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+// for basic account functionality. Any custom account type should extend this
+// type for additional functionality (e.g. vesting).
+message BaseAccount {
+  option (gogoproto.goproto_getters) = false;
+  option (gogoproto.goproto_stringer) = false;
+  option (gogoproto.equal) = false;
+
+  option (cosmos_proto.implements_interface) = "AccountI";
+
+  string address = 1;
```
Comment (araskachoi) [inline third_party/proto/cosmos/auth/v1beta1/auth.proto:20]: do we need `Coins` here too?

File: third_party/proto/cosmos/crisis/v1beta1/genesis.proto [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+option go_package = "github.com/cosmos/cosmos-sdk/x/crisis/types";
+
+import "gogoproto/gogo.proto";
+import "cosmos/base/v1beta1/coin.proto";
+
+// GenesisState defines the crisis module's genesis state.
+message GenesisState {
+  // constant_fee is the fee used to verify the invariant in the crisis
+  // module.
+  cosmos.base.v1beta1.Coin constant_fee = 3 [
```
Comment (araskachoi) [inline third_party/proto/cosmos/crisis/v1beta1/genesis.proto]: is this supposed to start at 1?

File: third_party/proto/cosmos/distribution/v1beta1/genesis.proto [hunk 10]
```diff
@@ -0,0 +1,197 @@
...
+
+  // validator_address is the address of the validator.
+  string validator_address = 1
+      [ (gogoproto.moretags) = "yaml:\"validator_address\"" ];
+  // height defines the block height at which the slash event occured.
+  uint64 height = 2;
+  // period is the period of the slash event.
+  uint64 period = 3;
+  // validator_slash_event describes the slash event.
+  ValidatorSlashEvent validator_slash_event = 4
```
Comment (araskachoi) [inline third_party/proto/cosmos/distribution/v1beta1/genesis.proto]: for proto, does this need the same exact variable name as the go-type or the name can be arbitrary as long as the types match?

Comment (araskachoi) [reply third_party/proto/cosmos/distribution/v1beta1/genesis.proto]: i see that the var name is `event` in the go defined struct

File: third_party/proto/cosmos/evidence/v1beta1/tx.proto [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+import "cosmos_proto/cosmos.proto";
+
+// MsgSubmitEvidence represents a message that supports submitting arbitrary
+// Evidence of misbehavior such as equivocation or counterfactual signing.
+message MsgSubmitEvidence {
+  option (gogoproto.equal) = false;
+  option (gogoproto.goproto_getters) = false;
+
+  string submitter = 1;
+  google.protobuf.Any evidence = 2
```
Comment (araskachoi) [inline third_party/proto/cosmos/evidence/v1beta1/tx.proto]: does the order matter here? in the go definition, it's `evidence` then `submitter`

File: third_party/proto/ibc/core/client/v1/genesis.proto [hunk 12]
```diff
@@ -0,0 +1,22 @@
+syntax = "proto3";
```
Comment (araskachoi) [inline third_party/proto/ibc/core/client/v1/genesis.proto]: do we need these ibc files or am i missing context and it lives in a different PR right now?

**Review by araskachoi — COMMENTED** [review]
Looks good to me, can you make some of the changes and maybe reply to my comments before I approve? thanks!

File: scripts/integration-test-all.sh (hunk 2 above)
Comment (fedekunze) [reply scripts/integration-test-all.sh:111]: just checked and the default is ok

File: third_party/proto/cosmos/auth/v1beta1/auth.proto (hunk 8 above)
Comment (fedekunze) [reply third_party/proto/cosmos/auth/v1beta1/auth.proto:20]: no, migrated to the bank module

File: third_party/proto/cosmos/crisis/v1beta1/genesis.proto (hunk 9 above)
Comment (fedekunze) [reply third_party/proto/cosmos/crisis/v1beta1/genesis.proto]: yup indeed, but this is from the SDK so we need to keep it like that

**Comment by fedekunze** [comment]
> Looks good to me, can you make some of the changes and maybe reply to my comments before I approve? 

Removed unnecessary third party proto files from the SDK @araskachoi. Also, regarding the scripts I'd prefer to fix that once the remaining PRs are approved

File: Makefile [hunk 13]
```diff
@@ -106,8 +106,7 @@ build_tags_comma_sep := $(subst $(whitespace),$(comma),$(build_tags))
 # process linker flags
 
 ldflags = -X github.com/cosmos/cosmos-sdk/version.Name=ethermint \
-		  -X github.com/cosmos/cosmos-sdk/version.ServerName=$(ETHERMINT_DAEMON_BINARY) \
-		  -X github.com/cosmos/cosmos-sdk/version.ClientName=$(ETHERMINT_CLI_BINARY) \
+		  -X github.com/cosmos/cosmos-sdk/version.AppName=$(ETHERMINT_BINARY) \
```
Comment (noot) [inline Makefile:109]: have ethermintcli and ethermintd been merged completely?

Comment (fedekunze) [reply Makefile:109]: yeah

**Review by noot — APPROVED** [review]

**Pull request closed without merging**